addons/binding/org.openhab.binding.chamberlainmyq/src/main/java/org/openhab/binding/chamberlainmyq/internal/HttpUtil.java

```diff
@@ -0,0 +1,201 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.chamberlainmyq.internal;
+
+import static org.openhab.binding.chamberlainmyq.ChamberlainMyQBindingConstants.USERAGENT;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.util.InputStreamContentProvider;
+import org.eclipse.jetty.http.HttpField;
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.http.HttpStatus;
+import org.eclipse.jetty.util.B64Code;
+import org.eclipse.jetty.util.StringUtil;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Need to Subclass to override the USERAGENT ID to spoof the MyQ App
+ * Some common methods to be used in both HTTP-In-Binding and HTTP-Out-Binding
+ *
+ * @author Scott Hanson - Initial contribution
+ * @author Thomas Eichstaedt-Engelen
+ * @author Kai Kreuzer - Initial contribution and API
+ * @author Svilen Valkanov - replaced Apache HttpClient with Jetty
+ */
+public class HttpUtil {
+
+    private static Logger logger = LoggerFactory.getLogger(HttpUtil.class);
+
+    private static HttpClient client = new HttpClient(new SslContextFactory());
```

Comment:
Please don't create your own `HttpClient`. Use the OSGi service HttpClientFactory via reference and let the factory create a client for you. The reason is that the framework need to be in control of open connections and ssl certificates.